File: Remove_Element/remove_element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        for j in range(len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i
    
# Example usage:
if __name__ == "__main__":
    solution = Solution()
    nums = [3, 2, 2, 3]
    val = 3
    new_length = solution.removeElement(nums, val)
    print(f"New length: {new_length}, Modified array: {nums}")

File: Remove_Element/remove_element.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution{
    public:
        int removeElement(vector<int>& nums, int val) {
            int i = 0;
            for (int j = 0; j < nums.size(); j++) {
                if (nums[j] != val) {
                    nums[i] = nums[j];
                    i++;
                }
            }
            return i;
        }
    };

int main() {
    Solution solution;
    vector<int> nums = {3, 2, 2, 3};
    int val = 3;
    int newLength = solution.removeElement(nums, val);
    cout << "New length: " << newLength << endl;
    cout << "Modified array: ";
    for (int i = 0; i < newLength; i++) {
        cout << nums[i] << " ";
    }
    
    return 0;
}